src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/addhearingoutcome/AddHearingOutcomeAboutToStartHandler.java

```diff
@@ -56,17 +61,33 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
             HearingsGetResponse response = hmcHearingsApiService.getHearingsRequest(Long.toString(caseDetails.getId()), HmcStatus.COMPLETED);
             List<CaseHearing> hmcHearings = response.getCaseHearings();
             if (!hmcHearings.isEmpty()) {
-                List<HearingDetails> selectedHearings = sscsCaseData.getHearings()
-                        .stream()
-                        .map(Hearing::getValue)
-                        .filter(value -> value.getStart() != null)
-                        .filter(value -> hmcHearings.stream()
-                                .anyMatch(hmcHearing -> Objects
-                                        .equals(hmcHearing.getHearingId().toString(), value.getHearingId())))
-                        .sorted(Comparator.comparing(HearingDetails::getStart).reversed())
-                        .toList();
+                Map<String, CaseHearing> hmcHearingsMap = hmcHearings.stream()
+                    .collect(Collectors.toMap(caseHearing -> caseHearing.getHearingId().toString(),
+                        caseHearing -> caseHearing));
+
+                if (sscsCaseData.getCompletedHearingsList() == null) {
+                    sscsCaseData.setCompletedHearingsList(new ArrayList<>());
+                }
+
+                sscsCaseData.getCompletedHearingsList().addAll(sscsCaseData.getHearings()
+                    .stream()
+                    .filter(value -> value.getValue().getStart() != null)
+                    .filter(hearing -> {
+                        CaseHearing hmcHearing = hmcHearingsMap.get(hearing.getValue().getHearingId());
+                        if (hmcHearing != null) {
+                            hearing.getValue().setStart(hearingUpdateService.convertUtcToUk(hmcHearing.getHearingDaySchedule().get(0).getHearingStartDateTime()));
```

Comment:
hmcHearing.getHearingDaySchedule().stream().findFirst().orElse(null) ?

Reply:
may be we can do same for all ? avoids null pointers and index out of bound exceptions

Reply:
updated them